python/lsst/sconsUtils/scripts.py

```diff
@@ -102,15 +109,26 @@ def initialize(cls, packageName, versionString=None, eupsProduct=None, eupsProdu
             except TypeError:
                 pass
             state.targets["version"] = state.env.VersionModule(versionModuleName)
+        scripts = []
         for root, dirs, files in os.walk("."):
             if "SConstruct" in files and root != ".":
                 dirs[:] = []
                 continue
             dirs[:] = [d for d in dirs if not d.startswith('.')]
             dirs.sort()  # happy coincidence that include < libs < python < tests
             if "SConscript" in files:
-                state.log.info("Using Sconscript at %s/SConscript" % root)
-                SConscript(os.path.join(root, "SConscript"))
+                scripts.append(os.path.join(root, "SConscript"))
+        if sconscriptOrder is None:
+            sconscriptOrder = ("lib", "python", "tests", "examples", "doc")
```

Comment:
I'll document that passing sconsOrder requires all SConscript directories to be passed; implementing a way to merge the default order with a user-supplied order is hard and possibly not well-defined in general.